lib/rules/no-async-in-computed-properties.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * @fileoverview Check if there are no async inside computed properties.
+ * @author Armano
+ */
+'use strict'
+
+const utils = require('../utils')
+
+const PROMISE_FUNCTIONS = [
+  'then',
+  'catch',
+  'finally'
+]
+
+const PROMISE_METHODS = [
+  'all',
+  'race',
+  'reject',
+  'resolve'
+]
+
+const TIMED_FUNCTIONS = [
+  'setTimeout',
+  'setInterval',
+  'setImmediate',
+  'requestAnimationFrame'
+]
+
+function isTimedFunction (node) {
+  return (
+    node.callee.type === 'Identifier' &&
+    TIMED_FUNCTIONS.indexOf(node.callee.name) !== -1
+  ) || (
+    node.callee.type === 'MemberExpression' &&
+    node.callee.object.name === 'window' && (
+      TIMED_FUNCTIONS.indexOf(node.callee.property.name) !== -1 ||
+      TIMED_FUNCTIONS.indexOf(node.callee.property.value) !== -1
+    )
+  ) && node.arguments.length
+}
+
+function isPromise (node) {
+  if (node.type === 'CallExpression' && node.callee && node.callee.type === 'MemberExpression') {
+    return ( // hello.PROMISE_FUNCTION()
+      node.callee.property &&
+      PROMISE_FUNCTIONS.indexOf(node.callee.property.name) !== -1
+    ) || ( // Promise.PROMISE_METHOD()
+      node.callee.object.type === 'Identifier' &&
+      node.callee.object.name === 'Promise' &&
+      PROMISE_METHODS.indexOf(node.callee.property.name) !== -1
+    ) || ( // somePromise.ANYTHING()
+      node.callee.object && isPromise(node.callee.object)
```

Comment:
I don't think so @armano2, in your example it would call `isPromise` with `memberExpression` -> `el.foo.bar` which will automatically return false. What you're checking is:
```
something.crazy.then().test()
```
But I don't see a real case with this kind of node. I don't think you'll ever end up with custom method after promise method so I think we can even remove this last condition. Add more tests if you want, delete it and check if they all pass.